src/main/java/net/minecraftforge/energy/EnergyStorage.java

```diff
@@ -29,6 +29,17 @@ public EnergyStorage(int capacity, int maxReceive, int maxExtract)
         this.maxReceive = maxReceive;
         this.maxExtract = maxExtract;
     }
+    
+    public void setEnergy(int energy)
+    {
+		this.energy = energy;
+
+		if (this.energy > capacity) {
+			this.energy = capacity;
+		} else if (this.energy < 0) {
+			this.energy = 0;
+		}
```

Comment:
Be careful not to use tabs, Forge uses spaces and so this is indented strangely.